internal/pkg/cache/cache.go

```diff
@@ -213,3 +217,21 @@ func refFromPackEntry(packEntry os.DirEntry) (ref string) {
 
 	return
 }
+
+// getGitHeadRef is a helper method that takes a directory which is a git
+// repository, and returns the SHA of the git HEAD of that repository.
+func getGitHeadRef(clonePath string) (string, error) {
+	sha := "n/a"
+	r, err := git.PlainOpen(clonePath)
+	if err != nil {
+		return sha, fmt.Errorf("could not read cloned repository: %v", err)
+	}
+	if r != nil {
```

Comment:
Can r be nil if an error is returned above?

Reply:
fair point!